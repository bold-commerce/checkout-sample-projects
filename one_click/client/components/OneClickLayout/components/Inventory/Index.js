export { default as inventory } from './Inventory'
export { default as InventoryItem } from './InventoryItem'