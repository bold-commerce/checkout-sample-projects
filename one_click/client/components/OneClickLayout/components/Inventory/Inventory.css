.Inventory__Main {
    padding: 16px 32px;
}

.Inventory__Message {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 16px;
    position: static;
    border: 1px solid rgba(0,0, 0, 0.2);
    line-height: 20px;
    margin-top: 16px;
    font-family: SF Pro Display;
    font-style: normal;
}

.Inventory__Title {
    color:rgba(0, 0, 0, 0.9);
    margin-bottom: 16px;
    font-size: 18px;
}

.Inventory__Description {
    font-size: 14px;
    color: rgba(0, 0, 0, 0.65);

}

.SoldOut {
    color: #B4002B;
    border: 1px solid #B4002B;
    background-color: #FFEBF0;
    padding: 4px 8px;
}

.Arrow {
    margin: 0px 8px;
    color: rgba(0, 0, 0, 0.65);
    font-size: x-large;
}

.InventoryItem {
    display: flex;
}
.InventoryItem__ProductDetails {
    padding: 0 20px;
    width: 100%;
}

.InventoryItem__ProductQuantityWrapper {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    margin-top: 4px;
}

.InventoryItem__ProductImageWrapper {
    width: 72px;
    height: 72px;
    padding: 3.6px;
    position: relative;
}

.InventoryItem__RemoveItem {
    width: 16px;
    height: 16px;
    border: none;
    background-color: inherit;
    cursor: pointer;
}

.ReturnLink {
    display: inline-block;
    line-height: 20px;
    font-size: 14px;
    margin-top: 16px;
}