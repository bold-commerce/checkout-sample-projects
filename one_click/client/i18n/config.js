import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;