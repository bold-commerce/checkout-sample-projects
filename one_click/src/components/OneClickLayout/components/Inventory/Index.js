export { default as Inventory } from './Inventory'
export { default as InventoryItem } from './InventoryItem'