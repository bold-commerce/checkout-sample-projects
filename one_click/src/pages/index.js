export * from './Confirmation';
export * from './IndexPage';
export * from './Processing';
export * from './Inventory';