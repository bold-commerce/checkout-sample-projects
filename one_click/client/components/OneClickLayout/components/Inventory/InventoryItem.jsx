import React from "react";
import PropTypes from 'prop-types';
import { Details, Image } from "@boldcommerce/stacks-ui/lib";
import './Inventory.css'
import { ArrowRight, TimesCircle } from "../Icons";

const InventoryItem = ({
    title, description, orderQty, stockQty, onRemove, image
}) => (
    <div className="SummaryBlock InventoryItem" >
        <div className="InventoryItem__ProductImageWrapper">
            <Image
                title={title}
                alt={title}
                src={image}
            />
        </div>
        <div className="InventoryItem__ProductDetails">
            <Details
                title={title}
                description={description}
            />
            <div className="InventoryItem__ProductQuantityWrapper">
                {
                    stockQty === 0 ?
                    <div className="SoldOut">Sold Out</div> :
                    <>                        
                        <div aria-label="ordered quantity" className="ProductQuantity"> 
                            {orderQty} 
                        </div>
                        <div 
                            className="Arrow"
                        >
                            <ArrowRight />
                        </div>
                        <div aria-label="available quantity" className="ProductQuantity">
                            {stockQty}
                        </div>
                    </>
                }
            </div>
        </div>
        <div className="InventoryItem__RemoveItemWrapper">
            {stockQty !== 0 &&
            <button 
                aria-label="remove item"
                className="InventoryItem__RemoveItem" 
                onClick={onRemove}
            >
                <TimesCircle />
            </button>
            }
        </div>
    </div>
);

InventoryItem.propTypes ={
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    orderQty: PropTypes.number.isRequired,
    stockQty: PropTypes.number.isRequired,
    onRemove: PropTypes.func,
    image: PropTypes.string,
};

export default InventoryItem;