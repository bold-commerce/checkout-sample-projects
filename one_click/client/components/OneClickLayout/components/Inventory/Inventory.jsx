import React, { useState, useContext } from "react";
import InventoryItem from "./InventoryItem";
import { useHistory, useLocation } from "react-router";
import { Button } from "@boldcommerce/stacks-ui"
import { Link } from "react-router-dom";
import { useLineItems } from "@boldcommerce/checkout-react-components";
import { AppContext } from '../../context/AppContext';
import { Header } from "../Header";
import './Inventory.css';

export const Inventory = ({}, ref) => {
    const { data: lineItems, updateLineItemQuantity, removeLineItem } = useLineItems();
    const [loading, setLoading] = useState(false);
    const { websiteName } = useContext(AppContext);
    const history = useHistory();
    const location = useLocation();
    const inventory = location.state;

    let invItems = lineItems.map((item) => {
        const product = inventory.find(i => i.id === item.product_data.product_id);
        item.product_data.stock = product.stock;
        item.product_data.inventory_issue = item.product_data.quantity > item.product_data.stock;
        if (item.product_data.inventory_issue) {
            return (
                <InventoryItem
                    key={item.product_data.line_item_key} 
                    title={product.name}
                    description={item.product_data.title}
                    orderQty={item.product_data.quantity}
                    stockQty={item.product_data.stock}
                    image={item.product_data.image_url}
                    onRemove={() => removeLineItem(item.product_data.line_item_key) }
                />);
        }
    })

    async function handleUpdateQuantities() {
        setLoading(true);
        const results = [];

        for (let i = 0; i < lineItems.length; i++) {
            const item = lineItems[i];
            if ( item.product_data.inventory_issue ) {
                if ( item.product_data.stock > 0 ) {
                    results.push(updateLineItemQuantity(item.product_data.line_item_key, item.product_data.stock));
                } else {
                    results.push(removeLineItem(item.product_data.line_item_key));
                }
            }
        }

        await Promise.all(results);
        setLoading(false);
        history.push("/");
    }

    return (
        <div ref={ref} className="Inventory__Main">
            <Header title={websiteName} />
            <div className="Inventory__Message">
                <div className="Inventory__Title">
                    Inventory issues
                </div>                
                <div className="Inventory__Description">
                    Some Products became unavailable and your cart has been updated, We're sorry for the inconvenience.
                </div>
            </div>
            {invItems}
            <div className="Inventory__footer">
                <Link className="ReturnLink" to="/">Return to cart</Link>
                <Button
                    onClick={handleUpdateQuantities}
                    className="CheckoutButton"
                    disabled={loading}
                >
                    Continue with changes
                </Button>
            </div>
        </div>
    )
}

const InventoryForwardedRef = React.forwardRef(Inventory);

export default InventoryForwardedRef;