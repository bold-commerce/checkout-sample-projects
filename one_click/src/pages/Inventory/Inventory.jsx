import React, { useState, useContext } from "react";
import { InventoryItem } from "../../components/InventoryItem";
import { useNavigate, useLocation, Link } from "react-router-dom";
import { Button } from "@boldcommerce/stacks-ui"
import { useLineItems } from "@boldcommerce/checkout-react-components";
import { useVariants } from "../../hooks";
import { AppContext } from '../../context/AppContext';
import { Header } from "../../components/Header";
import './Inventory.css';
import { useTranslation } from 'react-i18next';

export const Inventory = ({}, ref) => {
    const { data: lineItems, updateLineItemQuantity, removeLineItem } = useLineItems();
    const [loading, setLoading] = useState(false);
    const { websiteName } = useContext(AppContext);
    const handleVariants = useVariants();
    const navigate = useNavigate();
    const location = useLocation();
    const inventory = location.state;
    const { t } = useTranslation();

    let invItems = lineItems.map((item) => {
        const variantId = item.product_data.variant_id;
        item.product_data.stock = inventory[variantId].quantity;
        item.product_data.inventory_issue = inventory[variantId].inventory_tracker !== 'none' && item.product_data.quantity > item.product_data.stock;
        if (item.product_data.inventory_issue) {
            return (
                <InventoryItem
                    key={item.product_data.line_item_key} 
                    title={item.product_data.product_title}
                    variants={handleVariants(item.product_data.title)}
                    orderQty={item.product_data.quantity}
                    stockQty={item.product_data.stock}
                    image={item.product_data.image_url}
                    onRemove={() => removeLineItem(item.product_data.line_item_key) }
                />);
        }
    })

    async function handleUpdateQuantities() {
        setLoading(true);
        const results = [];

        for (let i = 0; i < lineItems.length; i++) {
            const item = lineItems[i];
            if ( item.product_data.inventory_issue ) {
                if ( item.product_data.stock > 0 ) {
                    results.push(updateLineItemQuantity(item.product_data.line_item_key, item.product_data.stock));
                } else {
                    results.push(removeLineItem(item.product_data.line_item_key));
                }
            }
        }

        await Promise.all(results);
        setLoading(false);
        navigate("/");
    }

    return (
        <div ref={ref} className="Inventory__Main">
            <Header title={websiteName} />
            <div className="Inventory__Message">
                <div className="Inventory__Title">
                    {t('inventory.issues')}
                </div>                
                <div className="Inventory__Description">
                    {t('inventory.issues_description')}
                </div>
            </div>
            {invItems}
            <div className="Inventory__footer">
                <Link to="/" className="ReturnLink" >{t('inventory.return_to_cart')}</Link>
                <Button
                    onClick={handleUpdateQuantities}
                    className="CheckoutButton"
                    disabled={loading}
                >
                    {t('inventory.continue')}
                </Button>
            </div>
        </div>
    )
}

const InventoryForwardedRef = React.forwardRef(Inventory);

export default InventoryForwardedRef;