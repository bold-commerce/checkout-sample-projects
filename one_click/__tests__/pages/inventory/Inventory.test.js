import React from 'react';
import { render } from '@testing-library/react';
import { Inventory } from '../../../src/pages/Inventory';
import { MemoryRouter } from 'react-router';
import { 
    exampleLineItems as MOCKexampleLineItems,
    exampleInventory as MOCKinventory
} from '../../utils/lineItemHelpers';
import '../../../src/i18n/config';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useLineItems: () => ({
        data: MOCKexampleLineItems,
        updateLineItemQuantity: (() => {}),
        removeLineItems: (() => {})
    })
})).mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useLocation: () => ({
        state: MOCKinventory
    })
})).mock('React', () => ({
    ...jest.requireActual('React'),
    useContext: () => ({
        websiteName: 'app.test'
    })
}));

describe('Inventory', () => {
    test('renders Inventory component', () => {
        const { asFragment } = render( 
            <Inventory />,
            {wrapper: MemoryRouter}
        );
        expect(asFragment()).toMatchSnapshot();
    })
})