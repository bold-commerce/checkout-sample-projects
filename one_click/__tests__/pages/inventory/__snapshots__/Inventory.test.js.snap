// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory renders Inventory component 1`] = `
<DocumentFragment>
  <div
    class="Inventory__Main"
  >
    <div
      class="Header"
    >
      <h1
        class="Header__Title"
      >
        app.test
      </h1>
      <button
        aria-label="Close"
        class="Header__Times"
      >
        <svg
          aria-labelledby="button"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="Inventory__Message"
    >
      <div
        class="Inventory__Title"
      >
        inventory.issues
      </div>
      <div
        class="Inventory__Description"
      >
        inventory.issues_description
      </div>
    </div>
    <div
      class="SummaryBlock InventoryItem"
    >
      <div
        class="CartItem__ImageDescriptionWrapper"
      >
        <div
          class="CartItem__ProductImageWrapper"
        >
          <div
            class="stx-image"
          >
            <div
              class="stx-image--content"
            >
              <div
                class="stx-image--holder"
              >
                <img
                  alt="Test title 2, Revenge of the title"
                  class="stx-image--image"
                  src=""
                  title="Test title 2, Revenge of the title"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="CartItem__ProductDetails"
        >
          <div
            class="stx-details"
          >
            <div
              class="stx-details__title"
            >
              Test title 2, Revenge of the title
            </div>
            <div
              class="stx-details__description"
            >
              <div>
                Number Two
              </div>
            </div>
          </div>
          <div>
            <div
              class="InventoryItem__ProductQuantityWrapper"
            >
              <div
                aria-label="ordered quantity"
                class="ProductQuantity"
              >
                2
              </div>
              <div
                class="Arrow"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    opacity="0.8"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M14.7471 4.33289L22.4142 12L14.7471 19.6671L13.3329 18.2529L19.5858 12L13.3329 5.7471L14.7471 4.33289Z"
                      fill="rgba(0, 0, 0, 0.65)"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3 11H21V13H3V11Z"
                      fill="rgba(0, 0, 0, 0.65)"
                      fill-rule="evenodd"
                    />
                  </g>
                </svg>
              </div>
              <div
                aria-label="available quantity"
                class="ProductQuantity"
              >
                1
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="InventoryItem__RemoveItemWrapper"
      >
        <button
          aria-label="remove item"
          class="InventoryItem__RemoveItem"
        >
          <svg
            aria-labelledby="button"
            fill="none"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              opacity="0.8"
            >
              <path
                clip-rule="evenodd"
                d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
                fill="black"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
                fill="black"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12Z"
                fill="black"
                fill-rule="evenodd"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      class="SummaryBlock InventoryItem"
    >
      <div
        class="CartItem__ImageDescriptionWrapper"
      >
        <div
          class="CartItem__ProductImageWrapper"
        >
          <div
            class="stx-image"
          >
            <div
              class="stx-image--content"
            >
              <div
                class="stx-image--holder"
              >
                <img
                  alt="Test title 3, the search for more money"
                  class="stx-image--image"
                  src=""
                  title="Test title 3, the search for more money"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="CartItem__ProductDetails"
        >
          <div
            class="stx-details"
          >
            <div
              class="stx-details__title"
            >
              Test title 3, the search for more money
            </div>
            <div
              class="stx-details__description"
            />
          </div>
          <div>
            <div
              class="InventoryItem__ProductQuantityWrapper"
            >
              <div
                class="SoldOut"
              >
                Sold Out
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="InventoryItem__RemoveItemWrapper"
      />
    </div>
    <div
      class="Inventory__footer"
    >
      <a
        class="ReturnLink"
        href="/"
      >
        inventory.return_to_cart
      </a>
      <button
        aria-busy="false"
        aria-disabled="false"
        class="stx-button CheckoutButton"
      >
        <span
          class="stx-button__text"
        >
          inventory.continue
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
